File: src/main/java/org/itmo/isLab1/dragons/DragonController.java
package org.itmo.isLab1.dragons;

import org.springframework.http.MediaType;
import org.springframework.web.bind.annotation.*;
import org.itmo.isLab1.common.framework.CrudController;
import org.itmo.isLab1.dragons.dto.*;

@RestController
@RequestMapping(value = "/api/dragons", produces = MediaType.APPLICATION_JSON_VALUE)
public class DragonController
    extends CrudController<
        Dragon,
        DragonDto,
        DragonCreateDto,
        DragonUpdateDto,
        DragonService> {

    public DragonController(
        DragonService service
    ) {
        super(service);
    }
}

File: frontend/src/controllers/DragonController.ts
import { DragonDto } from '@/interfaces/dto/dragons/DragonDto';
import { DragonCreateDto } from '@/interfaces/dto/dragons/DragonCreateDto';
import { DragonUpdateDto } from '@/interfaces/dto/dragons/DragonUpdateDto';
import CrudController from '@/interfaces/crud/CrudController';


export default new CrudController<DragonDto, DragonCreateDto, DragonUpdateDto>("dragons")